runtime/src/snapshot_utils/snapshot_storage_rebuilder.rs

```diff
@@ -317,10 +317,29 @@ impl SnapshotStorageRebuilder {
             })
             .collect::<Result<HashMap<AppendVecId, Arc<AccountStorageEntry>>, std::io::Error>>()?;
 
-        assert_eq!(slot_stores.len(), 1);
-        let (id, storage) = slot_stores.drain().next().unwrap();
-        self.storage
-            .insert(slot, AccountStorageReference { id, storage });
+        let storage = if slot_stores.len() > 1 {
+            let remapped_append_vec_id = self.next_append_vec_id.fetch_add(1, Ordering::AcqRel);
```

Comment:
Is `next_append_vec_id+1` the efficient way to assign the append_vec_id for the new storage? Will it push off the append_vec_id for the append_vecs for following slots? Would it result in more file renaming here?

https://github.com/solana-labs/solana/blob/master/runtime/src/serde_snapshot.rs#L624

Reply:
Well, I'm unhappy with this situation. The new ids start at 0:
https://github.com/solana-labs/solana/blob/db94e4564fefdf37d35362d42bfd9e3ade36261b/runtime/src/snapshot_utils.rs#L1258

We probably need to check for collisions in this new code similar to the code @HaoranYi linked...

Reply:
ok, checking for collisions and added a test

Reply:
lgtm